10_rest/willshen8/pkg/puppy/error.go

```diff
@@ -0,0 +1,54 @@
+package puppy
+
+import "fmt"
+
+type Error struct {
+	Message string  `json:"message"`
+	Code    ErrCode `json:"code"`
+}
+
+type ErrCode uint32
+
+func (e *Error) Error() string {
+	return fmt.Sprintf("Error code %d: %s",
+		e.Code, e.Message)
+}
+
+const (
+	ErrorCreateID ErrCode = iota
+	ErrorValueFormat
+	ErrorNegativeValue
+	ErrorReadPuppyValue
+	ErrorInvliadIDForUpdate
+	ErrorInvalidPuppyValueFormat
+	ErrorInvalidIDForDelete
+	ErrorInvalidInput
+	ErrorInvalidJSONInput
```

Comment:
Not ideal error codes, should be more like classifications. pretty sure I commented on this before?